TestingHelpers/MockFile.cs

```diff
@@ -410,7 +410,23 @@ public override void SetAccessControl(string path, FileSecurity fileSecurity)
 
         public override void SetAttributes(string path, FileAttributes fileAttributes)
         {
-            mockFileDataAccessor.GetFile(path).Attributes = fileAttributes;
+            var possibleFileData = mockFileDataAccessor.GetFile(path);
+            if (possibleFileData == null)
+            {
+                var directoryInfo = mockFileDataAccessor.DirectoryInfo.FromDirectoryName(path);
+                if (directoryInfo.Exists)
+                {
+                    directoryInfo.Attributes = fileAttributes;
+                }
+                else
+                {
+                    throw new DirectoryNotFoundException();
```

Comment:
according to my local "manual" tests it should throw a `FileNotFoundException` (à la `throw new FileNotFoundException(string.Format(CultureInfo.InvariantCulture, "Could not find file '{0}'.", path), path);` . see https://github.com/tathamoddie/System.IO.Abstractions/blob/master/TestingHelpers/MockFile.cs#L201